sync-core/src/test/java/com/cloudant/sync/query/AbstractQueryTestBase.java

```diff
@@ -17,27 +17,55 @@
 import static org.hamcrest.Matchers.containsInAnyOrder;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
 
+import com.cloudant.sync.datastore.DatastoreExtended;
+import com.cloudant.sync.datastore.DatastoreManager;
 import com.cloudant.sync.datastore.DocumentBodyFactory;
 import com.cloudant.sync.datastore.DocumentRevision;
 import com.cloudant.sync.datastore.MutableDocumentRevision;
+import com.cloudant.sync.sqlite.SQLDatabase;
+import com.cloudant.sync.util.TestUtils;
 
+import org.junit.After;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 
 import java.io.IOException;
 import java.sql.SQLException;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-public class QueryExecutorTest extends AbstractIndexTestBase {
+/**
+ *  The aim is to make sure that the post hoc matcher class behaves the
+ *  same as the SQL query engine.
+ *
+ *  To accomplish this, we test the entire test execution pipeline contained
+ *  within this class using {@link com.cloudant.sync.query.QueryExecutorSQLOnlyTest}
+ *  which exercises the SQL query engine matching functionality.  We then test
+ *  the same test execution pipeline contained within this class using
+ *  {@link com.cloudant.sync.query.QueryExecutorMatcherTest} which in turn
+ *  exercises the post hoc matcher matching functionality.
+ */
+public abstract class AbstractQueryTestBase {
```

Comment:
You still need a QueryExecutorTest class which tests the default implementations of IndexManager and QueryExecutor. Right now you only have the two classes which test each separately, you need the test for the use of both. The `QueryExecutorTest` might need some extra tests to cover the cases where a combination of both SQL and unindexed matcher is needed -- because the Abstract class probably only has tests which can be solved by either in isolation (depending on the indexes defined).